packages/atlas/src/MainLayout.tsx

```diff
@@ -59,6 +59,17 @@ export const MainLayout: FC = () => {
   const { trackPageView } = useSegmentAnalytics()
 
   useEffect(() => {
+    // had to include this timeout to make sure the page title is updated
+    setTimeout(
+      () =>
+        trackPageView(
+          document.title,
+          'viewer',
+          (location.pathname === absoluteRoutes.viewer.ypp() && searchParams.get('referrer')) || undefined
+        ),
+      1000
+    )
```

Comment:
maybe it would be easier to put this in `useHeadTags`? Don't know. If we want to keep this as it I would add cleanup function in the useEffect and clear timeout.

Reply:
not very obvious place to put this track function and too many extra renders there, so I've just added a destructor